### PR TITLE
Fix deprecation warnings for disabled rules

### LINKED ISSUE #6744: Never warn against deprecated and disabled rules
### What steps are needed to reproduce the bug?

N/A

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "at-rule-name-newline-after": null
}
```

### How did you run Stylelint?

stylelint .

### Which version of Stylelint are you using?

15.3.0

### What did you expect to happen?

No warning because that rule is expected not to be used

### What actually happened?

```log
Deprecation warnings:
 - The "at-rule-name-newline-after" rule is deprecated.
 - The "at-rule-name-space-after" rule is deprecated.
 - The "block-closing-brace-empty-line-before" rule is deprecated.
 - The "block-closing-brace-newline-after" rule is deprecated.
 - The "block-closing-brace-newline-before" rule is deprecated.
 - The "block-closing-brace-space-after" rule is deprecated.
 - The "block-closing-brace-space-before" rule is deprecated.
 - The "block-opening-brace-newline-after" rule is deprecated.
 - The "block-opening-brace-newline-before" rule is deprecated.
 - The "block-opening-brace-space-after" rule is deprecated.
 - The "block-opening-brace-space-before" rule is deprecated.
 - The "color-hex-case" rule is deprecated.
 - The "declaration-block-trailing-semicolon" rule is deprecated.
 - The "no-eol-whitespace" rule is deprecated.
 - The "number-leading-zero" rule is deprecated.
 - The "number-no-trailing-zeros" rule is deprecated.
 - The "selector-list-comma-newline-after" rule is deprecated.
 - The "selector-list-comma-newline-before" rule is deprecated.
 - The "selector-list-comma-space-after" rule is deprecated.
 - The "selector-list-comma-space-before" rule is deprecated.
 - The "selector-pseudo-class-case" rule is deprecated.
 - The "indentation" rule is deprecated.
```

### Does the bug relate to non-standard syntax?

_No response_

### Proposal to fix the bug

_No response_

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Closes #6744

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

**Review by mattxwang — APPROVED** [review]
LGTM - hopefully this new release helps users manage deprecations better!

**Comment by ybiquitous** [comment]
@mattxwang Thanks for the feedback.

> re: the title, wondering if there's a cleaner way to explain what this PR does - e.g. using the word "both". I don't think this is a huge concern though / not merge-blocking.

It could be helpful for users if we had a cleaner PR title (also a changelog entry?). 👍🏼 
Do you have a good suggestion?

**Comment by mattxwang** [comment]
Hm, nothing immediately jumps out at me. A handful of suggestions:

> Fix deprecation warnings for disabled rules

or

> Fix disabled rules from emitting deprecation warnings

or

> Fix deprecated and disabled rules from emitting warnings

It's certainly a challenging PR title to write 😅  any thoughts?

File: .changeset/poor-queens-battle.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed: deprecated and disabled rules not to warn
```
Comment (ybiquitous) [inline .changeset/poor-queens-battle.md:5]:
```suggestion
Fixed: deprecation warnings for disabled rules
```

**Comment by ybiquitous** [comment]
@mattxwang Thank you! I adopted your first suggestion. 👍🏼

**Pull request merged**